### ISSUE TITLE
Bug in gradient accumulation

### ISSUE THREAD

**Issue by hzhwcmhf** [thu-coai/seq2seq-pytorch#4]
https://github.com/thu-coai/seq2seq-pytorch/blob/51a40803f804bbdbdb601147cadee55ac954529d/seq2seq.py#L93

should be
```i % args.batch_num_per_gradient == 0```
